python-interface/src/MiniBotScripts/Buddybot.py

```diff
@@ -0,0 +1,190 @@
+import Note_Library as nl
+import live_audio_sample as las
+import random
+
+#Used for individual emotion testing 
+TEST_EMOTION = 0; #TODO remove once emotions finalized
+
+#Noise frequency thresholds #TODO check if correct
+SHOUT_THRESHOLD = 850
+SPEAK_THRESHOLD = 450
+
+# Should change at most by factor of 1 every second
+HAPPY_FACTOR = 10
+SAD_FACTOR = 200
+SURPRISE_FACTOR = 1
+ANGER_FACTOR = 50 #Currently not implemented
+
+SAMPLE_FILE = "sample.wav"
+
+#Global emotion counters
+bot_happy = 0
+bot_sad = 0
+bot_surprised = 0
+bot_angry = 0
+
+def randomEmotion():
+    return random.randint(0,3)
+
+def neutralNoise():
+    nl.playNote(nl.G5,0.15)
+
+def neutralNoise2():
+    nl.playNote(nl.G4,0.15)
+
+def happySong():
+    print("First note")        
+    nl.playNote(nl.G6,0.15)
+    print("Second note")
+    nl.playNote(nl.A6,0.15)
+    print("Third note")
+    nl.playNote(nl.G6,0.15)
+    nl.playNote(nl.A6,0.15)
+    nl.playNote(nl.B6,0.25)
+
+def showHappy(bot):
+    happySong()
```

Comment:
i guess it's too late now, but you could have made happy a bit more fun by having these execute concurrently